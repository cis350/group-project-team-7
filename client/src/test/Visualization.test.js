import React from 'react';
import { render, fireEvent, waitFor, screen } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import DataVisualization from '../components/Visualization';
import '@testing-library/jest-dom/extend-expect';
import { act } from 'react-dom/test-utils';
import { waitForElementToBeRemoved } from '@testing-library/dom';
// Mock the navigation
const mockNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockNavigate,
}));

describe('DataVisualization', () => {
  let consoleSpy;
  // Setup mock fetch in the global scope
  beforeEach(() => {
    global.fetch = jest.fn(() =>
        Promise.resolve({
            // ok: false, // Set ok property to false for unsuccessful response
            // status: 401, // Unauthorized response for the login attempt
            // json: () => Promise.resolve({ message: "Unauthorized" }),
            // text: () => Promise.resolve("No current user")
            ok: true,
            json: () => Promise.resolve([{ answer1: "10", answer2: "20", username: "user1" }]),
            text: () => Promise.resolve("Current user information")

        })
    );
    consoleSpy = jest.spyOn(console, 'error').mockImplementation(() => {});
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('renders the component and matches the snapshot', async () => {
    // Setup fetch to return some data
    global.fetch.mockResolvedValueOnce({
      ok: true,
      json: async () => [{ answer1: "10", answer2: "20", answer3: "30", username: "user1" }],
    });

    const { asFragment } = render(<Router><DataVisualization /></Router>);
    await waitFor(() => {
      expect(screen.getByText("Data Visualization - Bar Chart")).toBeInTheDocument();
    });

    expect(asFragment()).toMatchSnapshot();
  });

  it('fetches data and displays it correctly', async () => {
    // Return mock data for the fetch call
    global.fetch.mockResolvedValueOnce({
      ok: true,
      json: async () => [{ answer1: "Yes", answer2: "No", answer3: "Maybe", username: "testUser" }],
    });

    render(<Router><DataVisualization /></Router>);

    await waitFor(() => {
      expect(screen.getByText("testUser")).toBeInTheDocument();
    });
  });

  it('handles fetch errors gracefully', async () => {
    global.fetch.mockRejectedValueOnce(new Error('Failed to fetch data'));
    const { getByText } = render(
      <Router>
        <DataVisualization />
      </Router>
    );
    // doesn't crash
  });

    it('updates the charts when a new answer type is selected', async () => {
        global.fetch.mockResolvedValueOnce({
        ok: true,
        json: async () => [{ answer1: '10', answer2: '20', answer3: '30', username: 'User1' }]
        });

        render(
        <Router>
            <DataVisualization />
        </Router>
        );

        const initialItems = await screen.findAllByText('10');
        expect(initialItems.length).toBeGreaterThan(0); // Confirm initial data loaded
        act(() => {
            fireEvent.change(screen.getByLabelText('Select Answer Type:'), { target: { value: 'answer2' } });
        });
        const updatedItems = await screen.findAllByText('20');
        expect(updatedItems.length).toBeGreaterThan(0); // Confirm updated data is displayed
    });
});
