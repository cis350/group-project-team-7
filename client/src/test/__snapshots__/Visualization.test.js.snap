// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataVisualization renders the component and matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex justify-center items-center flex-wrap"
  >
    <div
      class="w-full sm:w-1/2 md:w-1/3 lg:w-1/3 p-4"
    >
      <h2
        class="text-lg font-semibold mb-4"
      >
        Data Visualization - Bar Chart
      </h2>
      <div
        class="mb-4"
      >
        <label
          class="mr-2 font-medium"
          for="answerSelector"
        >
          Select Answer Type:
        </label>
        <select
          class="px-3 py-2 border rounded-md"
          id="answerSelector"
        >
          <option
            value="answer1"
          >
            Answer 1
          </option>
          <option
            value="answer2"
          >
            Answer 2
          </option>
          <option
            value="answer3"
          >
            Answer 3
          </option>
        </select>
      </div>
      <div
        class="VictoryContainer"
        style="height: 100%; width: 100%; user-select: none; pointer-events: none; position: relative;"
      >
        <svg
          height="350"
          role="img"
          style="pointer-events: all; width: 100%; height: 100%;"
          viewBox="0 0 350 350"
          width="350"
        >
          <g
            role="presentation"
          >
            <line
              role="presentation"
              shape-rendering="auto"
              style="stroke: #90A4AE; fill: transparent; stroke-width: 2; stroke-linecap: round; stroke-linejoin: round;"
              vector-effect="non-scaling-stroke"
              x1="50"
              x2="300"
              y1="300"
              y2="300"
            />
            <g
              role="presentation"
            >
              <line
                role="presentation"
                shape-rendering="auto"
                style="stroke: #ECEFF1; fill: none; stroke-dasharray: 10, 5; stroke-linecap: round; stroke-linejoin: round; pointer-events: painted;"
                vector-effect="non-scaling-stroke"
                x1="96.02356406480118"
                x2="96.02356406480118"
                y1="300"
                y2="50"
              />
              <line
                role="presentation"
                shape-rendering="auto"
                style="stroke: #90A4AE; fill: transparent; size: 5px; stroke-width: 1; stroke-linecap: round; stroke-linejoin: round;"
                vector-effect="non-scaling-stroke"
                x1="96.02356406480118"
                x2="96.02356406480118"
                y1="300"
                y2="305"
              />
              <text
                direction="inherit"
                dx="0"
                id="chart-axis-0-tickLabels-0"
                x="96.02356406480118"
                y="323.26"
              >
                <tspan
                  dx="0"
                  dy="0"
                  style="font-family: 'Helvetica Neue', 'Helvetica', sans-serif; font-size: 12px; letter-spacing: normal; padding: 8px; fill: #455A64; stroke: transparent; stroke-width: 0;"
                  text-anchor="middle"
                  x="96.02356406480118"
                >
                  0.2
                </tspan>
              </text>
            </g>
            <g
              role="presentation"
            >
              <line
                role="presentation"
                shape-rendering="auto"
                style="stroke: #ECEFF1; fill: none; stroke-dasharray: 10, 5; stroke-linecap: round; stroke-linejoin: round; pointer-events: painted;"
                vector-effect="non-scaling-stroke"
                x1="142.04712812960236"
                x2="142.04712812960236"
                y1="300"
                y2="50"
              />
              <line
                role="presentation"
                shape-rendering="auto"
                style="stroke: #90A4AE; fill: transparent; size: 5px; stroke-width: 1; stroke-linecap: round; stroke-linejoin: round;"
                vector-effect="non-scaling-stroke"
                x1="142.04712812960236"
                x2="142.04712812960236"
                y1="300"
                y2="305"
              />
              <text
                direction="inherit"
                dx="0"
                id="chart-axis-0-tickLabels-1"
                x="142.04712812960236"
                y="323.26"
              >
                <tspan
                  dx="0"
                  dy="0"
                  style="font-family: 'Helvetica Neue', 'Helvetica', sans-serif; font-size: 12px; letter-spacing: normal; padding: 8px; fill: #455A64; stroke: transparent; stroke-width: 0;"
                  text-anchor="middle"
                  x="142.04712812960236"
                >
                  0.4
                </tspan>
              </text>
            </g>
            <g
              role="presentation"
            >
              <line
                role="presentation"
                shape-rendering="auto"
                style="stroke: #ECEFF1; fill: none; stroke-dasharray: 10, 5; stroke-linecap: round; stroke-linejoin: round; pointer-events: painted;"
                vector-effect="non-scaling-stroke"
                x1="188.0706921944035"
                x2="188.0706921944035"
                y1="300"
                y2="50"
              />
              <line
                role="presentation"
                shape-rendering="auto"
                style="stroke: #90A4AE; fill: transparent; size: 5px; stroke-width: 1; stroke-linecap: round; stroke-linejoin: round;"
                vector-effect="non-scaling-stroke"
                x1="188.0706921944035"
                x2="188.0706921944035"
                y1="300"
                y2="305"
              />
              <text
                direction="inherit"
                dx="0"
                id="chart-axis-0-tickLabels-2"
                x="188.0706921944035"
                y="323.26"
              >
                <tspan
                  dx="0"
                  dy="0"
                  style="font-family: 'Helvetica Neue', 'Helvetica', sans-serif; font-size: 12px; letter-spacing: normal; padding: 8px; fill: #455A64; stroke: transparent; stroke-width: 0;"
                  text-anchor="middle"
                  x="188.0706921944035"
                >
                  0.6
                </tspan>
              </text>
            </g>
            <g
              role="presentation"
            >
              <line
                role="presentation"
                shape-rendering="auto"
                style="stroke: #ECEFF1; fill: none; stroke-dasharray: 10, 5; stroke-linecap: round; stroke-linejoin: round; pointer-events: painted;"
                vector-effect="non-scaling-stroke"
                x1="234.0942562592047"
                x2="234.0942562592047"
                y1="300"
                y2="50"
              />
              <line
                role="presentation"
                shape-rendering="auto"
                style="stroke: #90A4AE; fill: transparent; size: 5px; stroke-width: 1; stroke-linecap: round; stroke-linejoin: round;"
                vector-effect="non-scaling-stroke"
                x1="234.0942562592047"
                x2="234.0942562592047"
                y1="300"
                y2="305"
              />
              <text
                direction="inherit"
                dx="0"
                id="chart-axis-0-tickLabels-3"
                x="234.0942562592047"
                y="323.26"
              >
                <tspan
                  dx="0"
                  dy="0"
                  style="font-family: 'Helvetica Neue', 'Helvetica', sans-serif; font-size: 12px; letter-spacing: normal; padding: 8px; fill: #455A64; stroke: transparent; stroke-width: 0;"
                  text-anchor="middle"
                  x="234.0942562592047"
                >
                  0.8
                </tspan>
              </text>
            </g>
            <g
              role="presentation"
            >
              <line
                role="presentation"
                shape-rendering="auto"
                style="stroke: #ECEFF1; fill: none; stroke-dasharray: 10, 5; stroke-linecap: round; stroke-linejoin: round; pointer-events: painted;"
                vector-effect="non-scaling-stroke"
                x1="280.1178203240059"
                x2="280.1178203240059"
                y1="300"
                y2="50"
              />
              <line
                role="presentation"
                shape-rendering="auto"
                style="stroke: #90A4AE; fill: transparent; size: 5px; stroke-width: 1; stroke-linecap: round; stroke-linejoin: round;"
                vector-effect="non-scaling-stroke"
                x1="280.1178203240059"
                x2="280.1178203240059"
                y1="300"
                y2="305"
              />
              <text
                direction="inherit"
                dx="0"
                id="chart-axis-0-tickLabels-4"
                x="280.1178203240059"
                y="323.26"
              >
                <tspan
                  dx="0"
                  dy="0"
                  style="font-family: 'Helvetica Neue', 'Helvetica', sans-serif; font-size: 12px; letter-spacing: normal; padding: 8px; fill: #455A64; stroke: transparent; stroke-width: 0;"
                  text-anchor="middle"
                  x="280.1178203240059"
                >
                  1
                </tspan>
              </text>
            </g>
          </g>
          <g
            role="presentation"
          >
            <line
              role="presentation"
              shape-rendering="auto"
              style="stroke: #90A4AE; fill: transparent; stroke-width: 2; stroke-linecap: round; stroke-linejoin: round;"
              vector-effect="non-scaling-stroke"
              x1="50"
              x2="50"
              y1="50"
              y2="300"
            />
            <g
              role="presentation"
            >
              <line
                role="presentation"
                shape-rendering="auto"
                style="stroke: #ECEFF1; fill: none; stroke-dasharray: 10, 5; stroke-linecap: round; stroke-linejoin: round; pointer-events: painted;"
                vector-effect="non-scaling-stroke"
                x1="50"
                x2="300"
                y1="253.9764359351988"
                y2="253.9764359351988"
              />
              <line
                role="presentation"
                shape-rendering="auto"
                style="stroke: #90A4AE; fill: transparent; size: 5px; stroke-width: 1; stroke-linecap: round; stroke-linejoin: round;"
                vector-effect="non-scaling-stroke"
                x1="50"
                x2="45"
                y1="253.9764359351988"
                y2="253.9764359351988"
              />
              <text
                direction="inherit"
                dx="0"
                id="chart-axis-1-tickLabels-0"
                x="37"
                y="258.2364359351988"
              >
                <tspan
                  dx="0"
                  dy="0"
                  style="font-family: 'Helvetica Neue', 'Helvetica', sans-serif; font-size: 12px; letter-spacing: normal; padding: 8px; fill: #455A64; stroke: transparent; stroke-width: 0;"
                  text-anchor="end"
                  x="37"
                >
                  0.2
                </tspan>
              </text>
            </g>
            <g
              role="presentation"
            >
              <line
                role="presentation"
                shape-rendering="auto"
                style="stroke: #ECEFF1; fill: none; stroke-dasharray: 10, 5; stroke-linecap: round; stroke-linejoin: round; pointer-events: painted;"
                vector-effect="non-scaling-stroke"
                x1="50"
                x2="300"
                y1="207.95287187039764"
                y2="207.95287187039764"
              />
              <line
                role="presentation"
                shape-rendering="auto"
                style="stroke: #90A4AE; fill: transparent; size: 5px; stroke-width: 1; stroke-linecap: round; stroke-linejoin: round;"
                vector-effect="non-scaling-stroke"
                x1="50"
                x2="45"
                y1="207.95287187039764"
                y2="207.95287187039764"
              />
              <text
                direction="inherit"
                dx="0"
                id="chart-axis-1-tickLabels-1"
                x="37"
                y="212.21287187039763"
              >
                <tspan
                  dx="0"
                  dy="0"
                  style="font-family: 'Helvetica Neue', 'Helvetica', sans-serif; font-size: 12px; letter-spacing: normal; padding: 8px; fill: #455A64; stroke: transparent; stroke-width: 0;"
                  text-anchor="end"
                  x="37"
                >
                  0.4
                </tspan>
              </text>
            </g>
            <g
              role="presentation"
            >
              <line
                role="presentation"
                shape-rendering="auto"
                style="stroke: #ECEFF1; fill: none; stroke-dasharray: 10, 5; stroke-linecap: round; stroke-linejoin: round; pointer-events: painted;"
                vector-effect="non-scaling-stroke"
                x1="50"
                x2="300"
                y1="161.9293078055965"
                y2="161.9293078055965"
              />
              <line
                role="presentation"
                shape-rendering="auto"
                style="stroke: #90A4AE; fill: transparent; size: 5px; stroke-width: 1; stroke-linecap: round; stroke-linejoin: round;"
                vector-effect="non-scaling-stroke"
                x1="50"
                x2="45"
                y1="161.9293078055965"
                y2="161.9293078055965"
              />
              <text
                direction="inherit"
                dx="0"
                id="chart-axis-1-tickLabels-2"
                x="37"
                y="166.18930780559648"
              >
                <tspan
                  dx="0"
                  dy="0"
                  style="font-family: 'Helvetica Neue', 'Helvetica', sans-serif; font-size: 12px; letter-spacing: normal; padding: 8px; fill: #455A64; stroke: transparent; stroke-width: 0;"
                  text-anchor="end"
                  x="37"
                >
                  0.6
                </tspan>
              </text>
            </g>
            <g
              role="presentation"
            >
              <line
                role="presentation"
                shape-rendering="auto"
                style="stroke: #ECEFF1; fill: none; stroke-dasharray: 10, 5; stroke-linecap: round; stroke-linejoin: round; pointer-events: painted;"
                vector-effect="non-scaling-stroke"
                x1="50"
                x2="300"
                y1="115.90574374079529"
                y2="115.90574374079529"
              />
              <line
                role="presentation"
                shape-rendering="auto"
                style="stroke: #90A4AE; fill: transparent; size: 5px; stroke-width: 1; stroke-linecap: round; stroke-linejoin: round;"
                vector-effect="non-scaling-stroke"
                x1="50"
                x2="45"
                y1="115.90574374079529"
                y2="115.90574374079529"
              />
              <text
                direction="inherit"
                dx="0"
                id="chart-axis-1-tickLabels-3"
                x="37"
                y="120.16574374079529"
              >
                <tspan
                  dx="0"
                  dy="0"
                  style="font-family: 'Helvetica Neue', 'Helvetica', sans-serif; font-size: 12px; letter-spacing: normal; padding: 8px; fill: #455A64; stroke: transparent; stroke-width: 0;"
                  text-anchor="end"
                  x="37"
                >
                  0.8
                </tspan>
              </text>
            </g>
            <g
              role="presentation"
            >
              <line
                role="presentation"
                shape-rendering="auto"
                style="stroke: #ECEFF1; fill: none; stroke-dasharray: 10, 5; stroke-linecap: round; stroke-linejoin: round; pointer-events: painted;"
                vector-effect="non-scaling-stroke"
                x1="50"
                x2="300"
                y1="69.88217967599412"
                y2="69.88217967599412"
              />
              <line
                role="presentation"
                shape-rendering="auto"
                style="stroke: #90A4AE; fill: transparent; size: 5px; stroke-width: 1; stroke-linecap: round; stroke-linejoin: round;"
                vector-effect="non-scaling-stroke"
                x1="50"
                x2="45"
                y1="69.88217967599412"
                y2="69.88217967599412"
              />
              <text
                direction="inherit"
                dx="0"
                id="chart-axis-1-tickLabels-4"
                x="37"
                y="74.14217967599413"
              >
                <tspan
                  dx="0"
                  dy="0"
                  style="font-family: 'Helvetica Neue', 'Helvetica', sans-serif; font-size: 12px; letter-spacing: normal; padding: 8px; fill: #455A64; stroke: transparent; stroke-width: 0;"
                  text-anchor="end"
                  x="37"
                >
                  1.0
                </tspan>
              </text>
            </g>
          </g>
          <g
            role="presentation"
          />
        </svg>
        <div
          style="z-index: 99; position: absolute; top: 0px; left: 0px; width: 100%; height: 100%;"
        >
          <svg
            height="350"
            style="overflow: visible; width: 100%; height: 100%;"
            viewBox="0 0 350 350"
            width="350"
          />
        </div>
      </div>
    </div>
    <div
      class="w-full sm:w-1/2 md:w-1/3 lg:w-1/3 p-4"
    >
      <h2
        class="text-lg font-semibold mb-4"
      >
        Data Visualization - Pie Chart
      </h2>
      <div
        class="VictoryContainer"
        style="width: 100%; height: 100%; pointer-events: none; position: relative;"
      >
        <svg
          height="400"
          role="img"
          style="pointer-events: all; width: 100%; height: 100%;"
          viewBox="0 0 400 400"
          width="400"
        >
          <g />
        </svg>
        <div
          style="z-index: 99; position: absolute; top: 0px; left: 0px; width: 100%; height: 100%;"
        >
          <svg
            height="400"
            style="overflow: visible; width: 100%; height: 100%;"
            viewBox="0 0 400 400"
            width="400"
          />
        </div>
      </div>
    </div>
    <div
      class="w-full sm:w-1/2 md:w-1/3 lg:w-1/3 p-4"
    >
      <h2
        class="text-lg font-semibold mb-4"
      >
        Most Frequent Usernames
      </h2>
      <div
        class="VictoryContainer"
        style="height: 100%; width: 100%; user-select: none; pointer-events: none; position: relative;"
      >
        <svg
          data-testid="bar-chart"
          height="350"
          role="img"
          style="pointer-events: all; width: 100%; height: 100%;"
          viewBox="0 0 350 350"
          width="350"
        >
          <g
            role="presentation"
          >
            <line
              role="presentation"
              shape-rendering="auto"
              style="stroke: #90A4AE; fill: transparent; stroke-width: 2; stroke-linecap: round; stroke-linejoin: round;"
              vector-effect="non-scaling-stroke"
              x1="50"
              x2="300"
              y1="300"
              y2="300"
            />
            <g
              role="presentation"
            >
              <line
                role="presentation"
                shape-rendering="auto"
                style="stroke: #ECEFF1; fill: none; stroke-dasharray: 10, 5; stroke-linecap: round; stroke-linejoin: round; pointer-events: painted;"
                vector-effect="non-scaling-stroke"
                x1="96.02356406480118"
                x2="96.02356406480118"
                y1="300"
                y2="50"
              />
              <line
                role="presentation"
                shape-rendering="auto"
                style="stroke: #90A4AE; fill: transparent; size: 5px; stroke-width: 1; stroke-linecap: round; stroke-linejoin: round;"
                vector-effect="non-scaling-stroke"
                x1="96.02356406480118"
                x2="96.02356406480118"
                y1="300"
                y2="305"
              />
              <text
                direction="inherit"
                dx="0"
                id="chart-axis-0-tickLabels-0"
                x="96.02356406480118"
                y="323.26"
              >
                <tspan
                  dx="0"
                  dy="0"
                  style="font-family: 'Helvetica Neue', 'Helvetica', sans-serif; font-size: 12px; letter-spacing: normal; padding: 8px; fill: #455A64; stroke: transparent; stroke-width: 0;"
                  text-anchor="middle"
                  x="96.02356406480118"
                >
                  0.2
                </tspan>
              </text>
            </g>
            <g
              role="presentation"
            >
              <line
                role="presentation"
                shape-rendering="auto"
                style="stroke: #ECEFF1; fill: none; stroke-dasharray: 10, 5; stroke-linecap: round; stroke-linejoin: round; pointer-events: painted;"
                vector-effect="non-scaling-stroke"
                x1="142.04712812960236"
                x2="142.04712812960236"
                y1="300"
                y2="50"
              />
              <line
                role="presentation"
                shape-rendering="auto"
                style="stroke: #90A4AE; fill: transparent; size: 5px; stroke-width: 1; stroke-linecap: round; stroke-linejoin: round;"
                vector-effect="non-scaling-stroke"
                x1="142.04712812960236"
                x2="142.04712812960236"
                y1="300"
                y2="305"
              />
              <text
                direction="inherit"
                dx="0"
                id="chart-axis-0-tickLabels-1"
                x="142.04712812960236"
                y="323.26"
              >
                <tspan
                  dx="0"
                  dy="0"
                  style="font-family: 'Helvetica Neue', 'Helvetica', sans-serif; font-size: 12px; letter-spacing: normal; padding: 8px; fill: #455A64; stroke: transparent; stroke-width: 0;"
                  text-anchor="middle"
                  x="142.04712812960236"
                >
                  0.4
                </tspan>
              </text>
            </g>
            <g
              role="presentation"
            >
              <line
                role="presentation"
                shape-rendering="auto"
                style="stroke: #ECEFF1; fill: none; stroke-dasharray: 10, 5; stroke-linecap: round; stroke-linejoin: round; pointer-events: painted;"
                vector-effect="non-scaling-stroke"
                x1="188.0706921944035"
                x2="188.0706921944035"
                y1="300"
                y2="50"
              />
              <line
                role="presentation"
                shape-rendering="auto"
                style="stroke: #90A4AE; fill: transparent; size: 5px; stroke-width: 1; stroke-linecap: round; stroke-linejoin: round;"
                vector-effect="non-scaling-stroke"
                x1="188.0706921944035"
                x2="188.0706921944035"
                y1="300"
                y2="305"
              />
              <text
                direction="inherit"
                dx="0"
                id="chart-axis-0-tickLabels-2"
                x="188.0706921944035"
                y="323.26"
              >
                <tspan
                  dx="0"
                  dy="0"
                  style="font-family: 'Helvetica Neue', 'Helvetica', sans-serif; font-size: 12px; letter-spacing: normal; padding: 8px; fill: #455A64; stroke: transparent; stroke-width: 0;"
                  text-anchor="middle"
                  x="188.0706921944035"
                >
                  0.6
                </tspan>
              </text>
            </g>
            <g
              role="presentation"
            >
              <line
                role="presentation"
                shape-rendering="auto"
                style="stroke: #ECEFF1; fill: none; stroke-dasharray: 10, 5; stroke-linecap: round; stroke-linejoin: round; pointer-events: painted;"
                vector-effect="non-scaling-stroke"
                x1="234.0942562592047"
                x2="234.0942562592047"
                y1="300"
                y2="50"
              />
              <line
                role="presentation"
                shape-rendering="auto"
                style="stroke: #90A4AE; fill: transparent; size: 5px; stroke-width: 1; stroke-linecap: round; stroke-linejoin: round;"
                vector-effect="non-scaling-stroke"
                x1="234.0942562592047"
                x2="234.0942562592047"
                y1="300"
                y2="305"
              />
              <text
                direction="inherit"
                dx="0"
                id="chart-axis-0-tickLabels-3"
                x="234.0942562592047"
                y="323.26"
              >
                <tspan
                  dx="0"
                  dy="0"
                  style="font-family: 'Helvetica Neue', 'Helvetica', sans-serif; font-size: 12px; letter-spacing: normal; padding: 8px; fill: #455A64; stroke: transparent; stroke-width: 0;"
                  text-anchor="middle"
                  x="234.0942562592047"
                >
                  0.8
                </tspan>
              </text>
            </g>
            <g
              role="presentation"
            >
              <line
                role="presentation"
                shape-rendering="auto"
                style="stroke: #ECEFF1; fill: none; stroke-dasharray: 10, 5; stroke-linecap: round; stroke-linejoin: round; pointer-events: painted;"
                vector-effect="non-scaling-stroke"
                x1="280.1178203240059"
                x2="280.1178203240059"
                y1="300"
                y2="50"
              />
              <line
                role="presentation"
                shape-rendering="auto"
                style="stroke: #90A4AE; fill: transparent; size: 5px; stroke-width: 1; stroke-linecap: round; stroke-linejoin: round;"
                vector-effect="non-scaling-stroke"
                x1="280.1178203240059"
                x2="280.1178203240059"
                y1="300"
                y2="305"
              />
              <text
                direction="inherit"
                dx="0"
                id="chart-axis-0-tickLabels-4"
                x="280.1178203240059"
                y="323.26"
              >
                <tspan
                  dx="0"
                  dy="0"
                  style="font-family: 'Helvetica Neue', 'Helvetica', sans-serif; font-size: 12px; letter-spacing: normal; padding: 8px; fill: #455A64; stroke: transparent; stroke-width: 0;"
                  text-anchor="middle"
                  x="280.1178203240059"
                >
                  1.0
                </tspan>
              </text>
            </g>
          </g>
          <g
            role="presentation"
          >
            <line
              role="presentation"
              shape-rendering="auto"
              style="stroke: #90A4AE; fill: transparent; stroke-width: 2; stroke-linecap: round; stroke-linejoin: round;"
              vector-effect="non-scaling-stroke"
              x1="50"
              x2="50"
              y1="50"
              y2="300"
            />
            <g
              role="presentation"
            >
              <line
                role="presentation"
                shape-rendering="auto"
                style="stroke: #ECEFF1; fill: none; stroke-dasharray: 10, 5; stroke-linecap: round; stroke-linejoin: round; pointer-events: painted;"
                vector-effect="non-scaling-stroke"
                x1="50"
                x2="300"
                y1="253.9764359351988"
                y2="253.9764359351988"
              />
              <line
                role="presentation"
                shape-rendering="auto"
                style="stroke: #90A4AE; fill: transparent; size: 5px; stroke-width: 1; stroke-linecap: round; stroke-linejoin: round;"
                vector-effect="non-scaling-stroke"
                x1="50"
                x2="45"
                y1="253.9764359351988"
                y2="253.9764359351988"
              />
              <text
                direction="inherit"
                dx="0"
                id="chart-axis-1-tickLabels-0"
                x="37"
                y="258.2364359351988"
              >
                <tspan
                  dx="0"
                  dy="0"
                  style="font-family: 'Helvetica Neue', 'Helvetica', sans-serif; font-size: 12px; letter-spacing: normal; padding: 8px; fill: #455A64; stroke: transparent; stroke-width: 0;"
                  text-anchor="end"
                  x="37"
                >
                  0.2
                </tspan>
              </text>
            </g>
            <g
              role="presentation"
            >
              <line
                role="presentation"
                shape-rendering="auto"
                style="stroke: #ECEFF1; fill: none; stroke-dasharray: 10, 5; stroke-linecap: round; stroke-linejoin: round; pointer-events: painted;"
                vector-effect="non-scaling-stroke"
                x1="50"
                x2="300"
                y1="207.95287187039764"
                y2="207.95287187039764"
              />
              <line
                role="presentation"
                shape-rendering="auto"
                style="stroke: #90A4AE; fill: transparent; size: 5px; stroke-width: 1; stroke-linecap: round; stroke-linejoin: round;"
                vector-effect="non-scaling-stroke"
                x1="50"
                x2="45"
                y1="207.95287187039764"
                y2="207.95287187039764"
              />
              <text
                direction="inherit"
                dx="0"
                id="chart-axis-1-tickLabels-1"
                x="37"
                y="212.21287187039763"
              >
                <tspan
                  dx="0"
                  dy="0"
                  style="font-family: 'Helvetica Neue', 'Helvetica', sans-serif; font-size: 12px; letter-spacing: normal; padding: 8px; fill: #455A64; stroke: transparent; stroke-width: 0;"
                  text-anchor="end"
                  x="37"
                >
                  0.4
                </tspan>
              </text>
            </g>
            <g
              role="presentation"
            >
              <line
                role="presentation"
                shape-rendering="auto"
                style="stroke: #ECEFF1; fill: none; stroke-dasharray: 10, 5; stroke-linecap: round; stroke-linejoin: round; pointer-events: painted;"
                vector-effect="non-scaling-stroke"
                x1="50"
                x2="300"
                y1="161.9293078055965"
                y2="161.9293078055965"
              />
              <line
                role="presentation"
                shape-rendering="auto"
                style="stroke: #90A4AE; fill: transparent; size: 5px; stroke-width: 1; stroke-linecap: round; stroke-linejoin: round;"
                vector-effect="non-scaling-stroke"
                x1="50"
                x2="45"
                y1="161.9293078055965"
                y2="161.9293078055965"
              />
              <text
                direction="inherit"
                dx="0"
                id="chart-axis-1-tickLabels-2"
                x="37"
                y="166.18930780559648"
              >
                <tspan
                  dx="0"
                  dy="0"
                  style="font-family: 'Helvetica Neue', 'Helvetica', sans-serif; font-size: 12px; letter-spacing: normal; padding: 8px; fill: #455A64; stroke: transparent; stroke-width: 0;"
                  text-anchor="end"
                  x="37"
                >
                  0.6
                </tspan>
              </text>
            </g>
            <g
              role="presentation"
            >
              <line
                role="presentation"
                shape-rendering="auto"
                style="stroke: #ECEFF1; fill: none; stroke-dasharray: 10, 5; stroke-linecap: round; stroke-linejoin: round; pointer-events: painted;"
                vector-effect="non-scaling-stroke"
                x1="50"
                x2="300"
                y1="115.90574374079529"
                y2="115.90574374079529"
              />
              <line
                role="presentation"
                shape-rendering="auto"
                style="stroke: #90A4AE; fill: transparent; size: 5px; stroke-width: 1; stroke-linecap: round; stroke-linejoin: round;"
                vector-effect="non-scaling-stroke"
                x1="50"
                x2="45"
                y1="115.90574374079529"
                y2="115.90574374079529"
              />
              <text
                direction="inherit"
                dx="0"
                id="chart-axis-1-tickLabels-3"
                x="37"
                y="120.16574374079529"
              >
                <tspan
                  dx="0"
                  dy="0"
                  style="font-family: 'Helvetica Neue', 'Helvetica', sans-serif; font-size: 12px; letter-spacing: normal; padding: 8px; fill: #455A64; stroke: transparent; stroke-width: 0;"
                  text-anchor="end"
                  x="37"
                >
                  0.8
                </tspan>
              </text>
            </g>
            <g
              role="presentation"
            >
              <line
                role="presentation"
                shape-rendering="auto"
                style="stroke: #ECEFF1; fill: none; stroke-dasharray: 10, 5; stroke-linecap: round; stroke-linejoin: round; pointer-events: painted;"
                vector-effect="non-scaling-stroke"
                x1="50"
                x2="300"
                y1="69.88217967599412"
                y2="69.88217967599412"
              />
              <line
                role="presentation"
                shape-rendering="auto"
                style="stroke: #90A4AE; fill: transparent; size: 5px; stroke-width: 1; stroke-linecap: round; stroke-linejoin: round;"
                vector-effect="non-scaling-stroke"
                x1="50"
                x2="45"
                y1="69.88217967599412"
                y2="69.88217967599412"
              />
              <text
                direction="inherit"
                dx="0"
                id="chart-axis-1-tickLabels-4"
                x="37"
                y="74.14217967599413"
              >
                <tspan
                  dx="0"
                  dy="0"
                  style="font-family: 'Helvetica Neue', 'Helvetica', sans-serif; font-size: 12px; letter-spacing: normal; padding: 8px; fill: #455A64; stroke: transparent; stroke-width: 0;"
                  text-anchor="end"
                  x="37"
                >
                  1.0
                </tspan>
              </text>
            </g>
          </g>
          <g
            role="presentation"
          />
        </svg>
        <div
          style="z-index: 99; position: absolute; top: 0px; left: 0px; width: 100%; height: 100%;"
        >
          <svg
            height="350"
            style="overflow: visible; width: 100%; height: 100%;"
            viewBox="0 0 350 350"
            width="350"
          />
        </div>
      </div>
    </div>
    <div
      class="w-full p-4"
    >
      <h2
        class="text-lg font-semibold mb-4"
      >
        Word Cloud
      </h2>
      <div>
        <svg
          preserveAspectRatio="xMinYMin meet"
          viewBox="0 0 500 100"
        >
          <g
            transform="translate(250,50)"
          >
            <text
              style="font-family: serif; font-style: italic; font-weight: bold; font-size: 0px; fill: #1f77b4;"
              text-anchor="middle"
              transform="translate(103,-13)rotate(1)"
            />
          </g>
        </svg>
      </div>
    </div>
    <button
      class="text-white w-[24vw] bg-blue-900 mt-4 hover:bg-blue-950 hover:scale-105 active:scale-100 duration-150 font-medium rounded px-5 py-4 focus:outline-none"
      type="button"
    >
      Return to Form
    </button>
  </div>
</DocumentFragment>
`;
